SU2_CFD/src/solvers/CGradientSmoothingSolver.cpp

```diff
@@ -909,28 +912,31 @@ void CGradientSmoothingSolver::ReadSensFromGeometry(const CGeometry* geometry) {
   }
 }
 
-void CGradientSmoothingSolver::WriteSensitivity(CGeometry* geometry, const CConfig* config, unsigned long val_marker) {
+void CGradientSmoothingSolver::WriteSensitivity(CGeometry* geometry, const CConfig* config) {
   unsigned long iPoint, total_index;
   unsigned int iDim;
   su2double* normal;
   su2double norm;
```

Comment:
Try to make things as local as possible please.

Reply:
What do you mean by local here?

I removed the marker BC_TYPE::NOT_AVAILABLE since it is no longer necessary.
Now the loop  
for (unsigned short iMarker = 0; iMarker < config->GetnMarker_All(); iMarker++) {
      if (config->GetMarker_All_DV(iMarker) == YES) {
should ensure that computations are only done if the current rank holds part of the design marker.

Reply:
What do you mean by local here?

I removed the additional boundary marker BC_TYPE::NOT_AVAILABLE and the associated checks. They are no longer necessary after I reworked the treatment of multiple dv markers.